mlir/lib/Conversion/VectorToArmSME/VectorToArmSME.cpp

```diff
@@ -356,6 +356,20 @@ struct TransposeOpToArmSMELowering
       return failure();
 
     auto loc = transposeOp.getLoc();
+    Value input = transposeOp.getVector();
+
+    if (auto xferOp = input.getDefiningOp<vector::TransferReadOp>()) {
+      // Fold transpose into transfer_read to enable in-flight transpose when
+      // converting to arm_sme.tile_load.
```

Comment:
not blocking: this feels like something that should be part of the transfer_write lowering, rather than the transpose lowering. I think we could just removing this allocating `vector.transpose` lowering (as we've never actually needed it).

Reply:
perhaps we might if we enable SME for transposes in IREE and they don't match this pattern?

Reply:
good point, we should update this lowering to use an extra tile then rather than remove.